sdks/python/src/opik/decorator/base_track_decorator.py

```diff
@@ -296,6 +309,16 @@ def _create_span(
         if current_span_data is not None:
             # There is already at least one span in current context.
             # Simply attach a new span to it.
+
+            if start_span_arguments.project_name != current_span_data.project_name:
+                LOGGER.warning(
```

Comment:
My concern here is that we might spam the user with these messages if they try to add `Spans` with different `project_name`s to a `Trace` with a specific project name, triggering a warning for each mismatch.

Reply:
Two options here, we can either move to "debug" and then users won't see it unless they enable debug mode or keep as is. Feel free to pick whichever one you think makes the most sense

Reply:
Ok, since this isn't a standard user flow, I suggest keeping a warning so the user knows when they're doing something wrong

Reply:
@japdubengsub I took a look, let's update the code so that is displayed only when the users explicitly specifies a project name for the span that is different than the parent span project name.

In the scenario below, we should not be showing a warning:

```python
import opik

@opik.track
def lower_level_function(input_arg: str):
    print(input_arg)


@opik.track(project_name="My project")
def top_level_function(input_arg: str):
    print(input_arg)
    lower_level_function(input_arg)

top_level_function("Hello world !")
```
We should not show the:
```
OPIK: You are attempting to log data into a nested span under the project name "None". However, the project name "My project" from parent span will be used instead.
```

Reply:
Agree, only when explicitly provided.